### PR TITLE
New render version

### PR DESCRIPTION
**Pull request by gas2serra**
I redesign the render inserting an image sub package that is close to the spec described in #275. Maybe, in the future, this package (or part of it) can become the standard image library of mcclim.
I add a section in the manual.

Alessandro

### CODE REVIEW

File: Backends/CLX-fb/port.lisp [hunk 1]
```diff
@@ -103,12 +121,12 @@
...
     (let ((event (xlib:process-event (clx-port-display port)
-				     :timeout 0.06
+				     ;;:timeout 0.06
 				     :handler #'clim-clx::event-handler :discard-p t)))
-      (maphash #'(lambda (key val)
-		   (when (typep key 'clx-fb-mirrored-sheet-mixin)
-		     (image-mirror-to-x (sheet-mirror key))))
-	       (slot-value port 'climi::sheet->mirror))
+      #+() (maphash #'(lambda (key val)
+                        (when (typep key 'clx-fb-mirrored-sheet-mixin)
```
Comment (dkochmanski) [inline Backends/CLX-fb/port.lisp]: if you want object comment, please use `#+(or)`

File: Extensions/render/mirror.lisp [hunk 2]
```diff
@@ -71,58 +75,93 @@
...
-    (let ((region
-	   (fill-image
-	    (image-mirror-image mirror)
-	    (make-rgba-design ink)
-	    image-mask
-	    :x x :y y :width width :height height
-	    :mask-dx x-dest :mask-dy y-dest)))
-      (%notify-image-updated mirror region))))
+  #+() (when (or (not (rectanglep clip-region))
+            (not (region-contains-region-p clip-region (make-rectangle* x y (+ x width) (+ y height)))))
```
Comment (dkochmanski) [inline Extensions/render/mirror.lisp]: for commenting out block of code, use `#+(or)`. Also - is it commented for a reason, or we could safely remove that code?

File: Extensions/render/mirror.lisp [hunk 3]
```diff
@@ -71,58 +75,93 @@
...
-
-
-(defmethod %draw-paths ((mirror image-mirror-mixin) paths transformation region ink background foreground)
-  (with-slots (render) mirror
-    (let ((region 
-	   (draw-paths render (image-mirror-image mirror) paths transformation region ink background foreground)))
-      (%notify-image-updated mirror region))))
+			x y width height ink background foreground clip-region)
+  #+() (when (or (not (rectanglep clip-region))
+            (not (region-contains-region-p clip-region (make-rectangle* x y (+ x width) (+ y height)))))
```
Comment (dkochmanski) [inline Extensions/render/mirror.lisp]: `#+(or)`

**Comment by gas2serra** [comment]
Hi Daniel,
Have you try it? What do you think?

**Comment by dkochmanski** [comment]
Hey, I haven't yet. I'll do that first thing tomorrow.

**Comment by gas2serra** [comment]
Thanks.

**Comment by dkochmanski** [comment]
Looks good. I have reported some issues, but these are most likely not regression. I'm merging it, thanks!

ps cube demo is way faster then previously. Great job.

**Comment by gas2serra** [comment]
Thanks.

**Pull request merged**